pkg/credentials/credentials.go

```diff
@@ -41,30 +41,96 @@ var (
 			MountPath:          "/credentials-gcp",
 			EnvCredentialsFile: common.GCPCredentialsEnvKey,
 			PluginName:         common.VeleroPluginForGCP,
-			PluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_GCP_PLUGIN_REPO"), os.Getenv("VELERO_GCP_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginMicrosoftAzure: {
 			IsCloudProvider:    true,
 			SecretName:         "cloud-credentials-azure",
 			MountPath:          "/credentials-azure",
 			EnvCredentialsFile: common.AzureCredentialsFileEnvKey,
 			PluginName:         common.VeleroPluginForAzure,
-			PluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_AZURE_PLUGIN_REPO"), os.Getenv("VELERO_AZURE_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginOpenShift: {
 			IsCloudProvider: false,
 			PluginName:      common.VeleroPluginForOpenshift,
-			PluginImage:     fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_REPO"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginCSI: {
 			IsCloudProvider: false,
 			//TODO: Check if the Registry needs to an upstream one from CSI
-			PluginName:  common.VeleroPluginForCSI,
-			PluginImage: fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_CSI_PLUGIN_REPO"), os.Getenv("VELERO_CSI_PLUGIN_TAG")),
+			PluginName: common.VeleroPluginForCSI,
 		},
 	}
 )
 
+func getPluginOverrideImage(velero *oadpv1alpha1.Velero) string {
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.AWSImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.AWSImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.OpenShiftImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.OpenShiftImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.AzureImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.AzureImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.CSIImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.CSIImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.GCPImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.GCPImageKey]
+	}
+	// TODO: handle better
+	return ""
+}
+
+func getPluginImage(velero *oadpv1alpha1.Velero) string {
```

Comment:
I would recommend this function instead takes in the name of the plugin. Then you don't need a loop, and you can have this function return the image for a given plugin name. Something like this:

```
func getPluginImage(pluginName string, velero *oadpv1alpha1.Velero) string {
    switch (pluginName) {
        case oadpv1alpha1.DefaultPluginAWS:
            return getAWSPluginImage(velero)
    }
...
}

func getAWSPluginImage(velero *oadpv1alpha1.Velero) string {
 	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.VeleroAWSPluginImageKey] != "" {
		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.VeleroAWSPluginImageKey]
	}
	if os.Getenv("VELERO_AWS_PLUGIN_REPO") == "" {
		return VeleroAWSPluginImage
	}
	return fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_AWS_PLUGIN_REPO"), os.Getenv("VELERO_AWS_PLUGIN_TAG"))
}
```

And then do this for each of the plugins, adding each condition to the switch statement